ambry-api/src/main/java/com.github.ambry/account/AccountService.java

```diff
@@ -87,4 +87,19 @@
    * @return A collection of {@link Account}s.
    */
   public Collection<Account> getAllAccounts();
+
+  /**
+   * Adds an {@link AccountUpdateListener} for newly created or updated {@link Account}s.
+   * @param listener The {@link AccountUpdateListener} to add.
+   * @return {@code true} if the specified {@link AccountUpdateListener} has already been added, {@code false}
+   *                      otherwise.
+   */
+  public boolean addListener(AccountUpdateListener listener);
```

Comment:
Instead of making an `AccountUpdateListener` class, how do you feel about this method taking a `Consumer<Collection<Account>>` (part of the java stdlib)?